codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/PaginatedOperationWithResultKeyPaginator.java

```diff
@@ -0,0 +1,97 @@
+ *
+ *
+ * 2) Using For loop for (PaginatedOperationWithResultKeyResponse response : responses) { // do something; }
+ *
+ *
+ * 3) Use iterator directlyresponses.iterator().forEachRemaining(.....);
+ *
+ *
+ * Note: If you prefer to have control on service calls, use the
+ * {@link JsonProtocolTestsClient#paginatedOperationWithResultKey(PaginatedOperationWithResultKeyRequest)} API.
+ */
+@Generated("software.amazon.awssdk:codegen")
+public final class PaginatedOperationWithResultKeyPaginator implements SdkIterable<PaginatedOperationWithResultKeyResponse> {
+    private final JsonProtocolTestsClient client;
+
+    private final PaginatedOperationWithResultKeyRequest firstRequest;
+
+    private final NextPageFetcher nextPageFetcher;
+
+    public PaginatedOperationWithResultKeyPaginator(final JsonProtocolTestsClient client,
+                                                    final PaginatedOperationWithResultKeyRequest firstRequest) {
+        this.client = client;
+        this.firstRequest = firstRequest;
+        this.nextPageFetcher = new PaginatedOperationWithResultKeyResponseFetcher();
+    }
+
+    @Override
+    public Iterator<PaginatedOperationWithResultKeyResponse> iterator() {
+        return new PaginatedResponsesIterator(nextPageFetcher);
+    }
+
+    /**
+     * Returns an iterable to iterate through the paginated {@link PaginatedOperationWithResultKeyResponse#items()}
+     * member. The returned iterable is used to iterate through the results across all response pages and not a single
+     * page.
+     *
+     * This method is useful if you are interested in iterating over the paginated member in the response pages instead
+     * of the top level pages. Similar to iteration over pages, this method internally makes service calls to get the
+     * next list of results until the iteration stops or there are no more results.
+     */
+    public SdkIterable<SimpleStruct> items() {
+        Function<PaginatedOperationWithResultKeyResponse, Iterator<SimpleStruct>> getIterator = response -> response != null ? response
+            .items().iterator() : null;
+
+        return new PaginatedItemsIterable(this, getIterator);
+    }
+
+    private class PaginatedOperationWithResultKeyResponseFetcher implements
+                                                                 NextPageFetcher<PaginatedOperationWithResultKeyResponse> {
+        @Override
+        public boolean hasNextPage(PaginatedOperationWithResultKeyResponse oldPage) {
+            return oldPage.nextToken() != null;
+        }
+
+        @Override
+        public PaginatedOperationWithResultKeyResponse nextPage(PaginatedOperationWithResultKeyResponse oldPage) {
+            if (oldPage == null) {
+                return client.paginatedOperationWithResultKey(firstRequest);
+            } else {
+                return client.paginatedOperationWithResultKey(firstRequest.toBuilder().nextToken(oldPage.nextToken()).build());
```

Comment:
Minor but perhaps just a ternary here.

```java
return client.paginatedOperationWithResultKey(firstRequest.toBuilder()
      .nextToken(oldPage == null ? null : oldPage.nextToken())
     .build();
```

Reply:
The output_token is a list i.e., there might be multiple values in response that is needed to get the next page. I feel ternary becomes convoluted to read and i prefer to keep the if-else. 
```
return client.paginatedOperationWithResultKey(firstRequest.toBuilder()
      .nextToken1(oldPage == null ? null : oldPage.nextToken1())
      .nextToken2(oldPage == null ? null : oldPage.nextToken2())
     .build();
```

Reply:
Minor, if using `if-else`, It might be cleaner to remove the else.

```
if (oldPage == null) {
      return client.paginatedOperationWithResultKey(firstRequest);
}

return client.paginatedOperationWithResultKey(firstRequest.toBuilder().nextToken(oldPage.nextToken()).build());
```